### PR TITLE
Add `frame width` option to default shapes: circle/rectlange/square 

### PR DESCRIPTION
**Pull request by simoncarrignon**
Related to these very old quesions on stackoverflow : https://stackoverflow.com/questions/18483830/vertex-frame-width-in-r-network-plot https://stackoverflow.com/questions/9759484/vertex-border-color-width-in-r-graph-plot
And using the code @gaborcsardi  wrote here: https://lists.gnu.org/archive/html/igraph-help/2013-03/msg00030.html
I updated the default shapes `circle`, `square`, `rectangle` to use a new parameter `frame.width` to change the `lwd` R parameter used in the function `symbols` to define the line width of the symbol drawn.

```{r}
library(igraph)
n  <-  20
g <- make_ring(n)
g$layout <- layout_in_circle
V(g)$size <- sample(10:30,n,replace=T)
#one width
plot(g,vertex.color=rainbow(n,alpha=.6)[1:n],vertex.frame.width=5)
```
image

```{r}
#multiple width
V(g)$frame.width <- seq(2,10,length.out=n)
plot(g,vertex.color=rainbow(n,alpha=.6)[1:n])
```
image

```{r}
# Using square shape, growing size 
V(g)$shape="square"
V(g)$size=seq(10,50,length.out=n)
#one width
plot(g,vertex.color=rainbow(n,alpha=.6)[1:n],vertex.frame.width=2)
```
image

```{r}
#multiple frame width
V(g)$frame.width <- seq(2,10,length.out=n)
plot(g,vertex.color=rainbow(n,alpha=.6)[1:n])
```

image

```{r}
# Using rectangle shape, changing size 
V(g)$shape="rectangle"
V(g)$size=seq(10,50,length.out=n)
V(g)$size2=rev(seq(10,50,length.out=n))
#one width
plot(g,vertex.frame.color=rainbow(n)[1:n],vertex.color=rev(rainbow(n,alpha=.6)[1:n]),vertex.frame.width=5)
```

image

```{r}
#multiple frame width
V(g)$frame.width <- rev(seq(2,10,length.out=n))
plot(g,vertex.color=rainbow(n,alpha=.6)[1:n])
```
image

### CODE REVIEW

**Comment by ntamas** [comment]
This looks great, thanks a lot! It will be part of release 1.3 if everything goes well. Let me know if you need this for your own work and you are not comfortable with keeping a patched version around, and I'll look into making a 1.2.x release earlier (1.3 is still a few weeks ahead).

**Comment by simoncarrignon** [comment]
No worries I'm totally fine using my patched version for now!

**Comment by ntamas** [comment]
Okay, great! I'm waiting for the CI tests to pass and then I'll merge into 1.3.

File: R/plot.shapes.R [hunk 1]
```diff
@@ -505,14 +519,27 @@ add_shape <- function(shape, clip=shape_noclip,
...
-          squares=2*vertex.size, add=TRUE, inches=FALSE)
+  if (length(vertex.frame.width) ==1) {
+    symbols(x=coords[,1], y=coords[,2], bg=vertex.color, fg=vertex.frame.color,
+          squares=2*vertex.size, lwd=vertex.frame.width, add=TRUE, inches=FALSE)
+  } else {
+    mapply(coords[,1], coords[,2], vertex.color, vertex.frame.color,
+           vertex.size, vertex.frame.width,
+           FUN=function(x, y, bg, fg, size, lwd) {
+               symbols(x=x, y=y, bg=bg, fg=fg, lwd=lwd,
+                       square=2*size, add=TRUE, inches=FALSE)
```
Comment (ntamas) [inline R/plot.shapes.R:540]: I think this should be `squares` here, not `square`. (The CI warns about this in a `NOTE`).

Comment (simoncarrignon) [reply R/plot.shapes.R]: just saw this, sorry bout that ;)

**Pull request merged**